src/WooCommerce.php

```diff
@@ -249,6 +249,25 @@ public static function cron_remove_back_in_stock() {
     }
   }
 
+  /**
+   * Cron event callback to remove outdated back-in-stock product metadata.
+   */
+  public static function cron_orphan_variants_cleanup() {
+    global $wpdb;
+    $orphan_variants_ids = $wpdb->get_col(
+      "SELECT products.ID products
+      FROM {$wpdb->posts} products
+      LEFT JOIN {$wpdb->posts} wp ON wp.ID = products.post_parent
+      WHERE wp.ID IS NULL AND products.post_type = 'product_variation';"
```

Comment:
Did you run this query?  There is a superfluous "products" in the selected fields, which is a syntax/parse error.
```suggestion
      "SELECT p.ID
      FROM {$wpdb->posts} p
      LEFT JOIN {$wpdb->posts} parent ON parent.ID = p.post_parent
      WHERE parent.ID IS NULL AND p.post_type = 'product_variation'"
```

Reply:
@sun , Yes, of course. I never submit coding without running/testing it.  :)

I agree that the products is superfluous, but it doesn't trigger any error: nevertheless 
<img width="269" alt="image" src="https://user-images.githubusercontent.com/18492002/183717738-1b8bc583-fe80-4a64-83b8-f13c11062f0f.png">

I've been using exactly the same query provided by WooCommerce: https://github.com/woocommerce/woocommerce/blob/3a76aa42555f037a0a565e9f0feaa43f89b197d2/plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-system-status-tools-v2-controller.php#L464